From c37140662d22e86e1096c953b888b6496b7e1f64 Mon Sep 17 00:00:00 2001
From: Andrei Vagin <avagin@gmail.com>
Date: Tue, 21 Dec 2021 21:59:13 -0800
Subject: [PATCH 094/120] uffd: call disconnect_from_page_server to shutdown a
 page-server connection

We need to be sure that page-server doesn't wait for a new command when we
call gnutls_bye() that sends an alert containing a close request.

Signed-off-by: Andrei Vagin <avagin@gmail.com>
---
 criu/uffd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/criu/uffd.c b/criu/uffd.c
index f01e6999b..45ac8ba77 100644
--- a/criu/uffd.c
+++ b/criu/uffd.c
@@ -1468,7 +1468,7 @@ int cr_lazy_pages(bool daemon)
 
 	ret = handle_requests(epollfd, &events, nr_fds);
 
-	tls_terminate_session();
+	disconnect_from_page_server();
 
 	xfree(events);
 	return ret;
-- 
2.34.1

